py/tests/rest/htm_rest_api_test.py

```diff
@@ -0,0 +1,208 @@
+    self.assertEqual(r, 'OK')
+
+    # Delete a region
+    r = net.delete_region('tm')
+    self.assertEqual(r, 'OK')
+
+    # Delete a Network
+    r = net.delete_all()
+    self.assertEqual(r, 'OK')
+
+  def testNetworkRESTExample(self):
+    rsp = requests.get(HOST + '/hi')
+    self.assertEqual(rsp.status_code, requests.codes.ok)
+    self.assertEqual(rsp.text, 'Hello World!\n')
+
+    net = NetworkREST(host=HOST, verbose=True)
+
+    encoder = net.add_region('encoder', 'RDSEEncoderRegion', {
+      'size': 1000,
+      'sparsity': 0.2,
+      'radius': 0.03,
+      'seed': 2019,
+      'noise': 0.01
+    })
+
+    sp = net.add_region('sp', 'SPRegion', {
+      'columnCount': 2048,
+      'globalInhibition': True
+    })
+
+    tm = net.add_region('tm', 'TMRegion', {
+      'cellsPerColumn': 8,
+      'orColumnOutputs': True
+    })
+
+    link1 = net.add_link(encoder, sp, 'encoded', 'bottomUpIn')
+    link2 = net.add_link(sp, tm, 'bottomUpOut', 'bottomUpIn')
+
+    net.create()
+
+    r = tm.param('cellsPerColumn')
+    self.assertEqual(r, '8')
+    # iterate EPOCHS times
+    x = 0.00
+    for e in range(EPOCHS):
+      # Data is a sine wave,    (Note: first iteration is for x=0.01, not 0)
+      x += 0.01  # advance one step, 0.01 radians
+      s = math.sin(x)
+
+      # Send set parameter message to feed "sensedValue" parameter data into RDSE encoder for this iteration.
+      r = encoder.param('sensedValue', '{:.2f}'.format(s))
+      self.assertEqual(r, 'OK')
+
+      # Execute one iteration of the Network object
+      r = net.run(1)
+      self.assertEqual(r, 'OK')
+
+    # Retreive the final anomaly score from the TM object's 'anomaly' output.
+    score = tm.output('anomaly')
+    self.assertEqual(score, [1])
+
+    print('Anomaly Score: ' + str(score))
+    #Note: Anomaly score will be 1 until there have been enough iterations to 'learn' the pattern.
+
+  def testNetworkRESTDelete(self):
+    net = NetworkREST(host=HOST, verbose=True)
+
+    encoder = net.add_region('encoder', 'RDSEEncoderRegion', {
+      'size': 1000,
+      'sparsity': 0.2,
+      'radius': 0.03,
+      'seed': 2019,
+      'noise': 0.01
+    })
+
+    sp = net.add_region('sp', 'SPRegion', {
+      'columnCount': 2048,
+      'globalInhibition': True
+    })
+
+    tm = net.add_region('tm', 'TMRegion', {
+      'cellsPerColumn': 8,
+      'orColumnOutputs': True
+    })
+
+    link1 = net.add_link(encoder, sp, 'encoded', 'bottomUpIn')
+    link2 = net.add_link(sp, tm, 'bottomUpOut', 'bottomUpIn')
+
+    net.create()
+
+    # Now delete the second link.
+    r = link2.destory()
+    self.assertEqual(r, 'OK')
+
+    # Delete a region
+    r = tm.destory()
+    self.assertEqual(r, 'OK')
+
+    # Delete a Network
+    r = net.delete_all()
+    self.assertEqual(r, 'OK')
+
+  def tearDown(self):
+    try:
+      r = requests.get('{}/stop'.format(HOST))
+      print(r)
+    except:
+      pass
+    self._thread.join()
```

Comment:
I think it is hanging here.
Is there a way to kill another thread in Python?
If you should get an exception while sending the "/stop" message or the server does not handle it for any reason, the thread.join( ) would wait forever.

Reply:
Oh, you need to remove the line "self._thread.join( )."

Reply:
should be good now?